File: slog-composor/src/main/java/net/slog/composor/ComposorLoggerFactory.java
package net.slog.composor;

import net.slog.ILoggerFactory;
import net.slog.SLogBinder;

/**
 * Composor实现
 * Created by zhongyongsheng on 2018/3/19.
 */

public class ComposorLoggerFactory implements ILoggerFactory {
    @Override
    public SLogBinder.SLogBindLogger getLogger(Class cls) {
        return new LogComposor(cls.getSimpleName());
    }

    @Override
    public SLogBinder.SLogBindLogger getLogger(String name) {
        return new LogComposor(name);
    }
}

File: slog-composor/src/main/java/net/slog/composor/ComposorLoggerFactory.kt
package net.slog.composor

import net.slog.ILoggerFactory
import net.slog.SLogBinder

/**
 * Composor实现
 * Created by zhongyongsheng on 2018/3/19.
 */

class ComposorLoggerFactory : ILoggerFactory {
    override fun getLogger(cls: Class<*>): SLogBinder.SLogBindLogger {
        return ComposorLogBindLogger(cls.simpleName)
    }

    override fun getLogger(name: String): SLogBinder.SLogBindLogger {
        return ComposorLogBindLogger(name)
    }
}
